autocodesign/projectmanager/projectmanager.go

```diff
@@ -17,13 +17,28 @@ type Project struct {
 	projHelper ProjectHelper
 }
 
+// Factory ...
+type Factory struct {
+	params InitParams
+}
+
 // InitParams ...
 type InitParams struct {
 	ProjectOrWorkspacePath string
 	SchemeName             string
 	ConfigurationName      string
 }
 
+// NewFactory ...
+func NewFactory(params InitParams) Factory {
```

Comment:
why is this factory needed?

Reply:
Used this as a tool to allow for passing in the project unopened; parsing the project only when needed.